en/docs/wip/change-developer-portal-password.md

```diff
@@ -1,64 +1,95 @@
-
-     [![Change Developer Portal password User Menu]({{base_path}}/assets/img/learn/change-devportal-password-user-menu-click.png)]({{base_path}}/assets/img/learn/change-devportal-password-user-menu-click.png)    
-
-3. Enter your old password, a new password, re-enter the new password, and thereafter click **SAVE** to submit changes.
-
-     [![Change Developer Portal password Save Clicking]({{base_path}}/assets/img/learn/change-devportal-password-policy-guideline-display.png)]({{base_path}}/assets/img/learn/change-devportal-password-policy-guideline-display.png)
+# Change the Developer Portal Password  
+  
+  You can change your own password for the WSO2 API Manager Developer Portal using the Developer Portal UI.    
+  
+Follow the instructions below to change your password:   
+  
+1. Sign in to the Developer Portal.      
+  
+     `https://<hostname>:9443/devportal`   
+  Example: `https://localhost:9443/devportal`   
+  Use your username and password to sign in.      
+  
+2. Click  on your username at the top right corner and select  **Change Password**.    
+  
+     [![Change Developer Portal password User Menu]({{base_path}}/assets/img/learn/change-devportal-password-user-menu-click.png)]({{base_path}}/assets/img/learn/change-devportal-password-user-menu-click.png)      
+  
+3. Enter your old password, a new password, re-enter the new password, and thereafter click **SAVE** to submit changes.  
+  
+     [![Developer portal password change submit]({{base_path}}/assets/img/learn/change-devportal-password-submiting.png)]({{base_path}}/assets/img/learn/change-devportal-password-submiting.png)  
+  
+## Use custom password policy
+You can define your custom password policy by defining one or both of the followings. 
+    
+1. [User store password regex](https://apim.docs.wso2.com/en/latest/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store/#configuring-a-jdbc-user-store)  
+Example:
+```toml
+[user_store]
+type = "database_unique_id"
+
+[user_store.properties]
+PasswordJavaRegEx = "^[\\S]{6,30}$"
+```
+2. [Identity management  password policies](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/identity-management-for-the-api-dev-portal/#password-policies)
+
+<html>    
+    <div class="admonition note">    
+        <p class="admonition-title">Note</p>    
+        <p>
+            The password policy set by identity management password policies can be overridden by Management console for each tenant.
+        </p>    
+    </div>
+</html>
+<html>    
+    <div class="admonition note">    
+        <p class="admonition-title">Note</p>    
+        <p>
+            When changing the password, new password will be validated against all custom password policies. If the password policies are set to be in conflict of each other you may not be able to change your password.
```

Comment:
Can we configure mutiple "custom password policies" ?

Reply:
I think u wanted to mean also the user store manager password pattern. Better to say as it.. "both password policies and user store manager password pattern ".

Reply:
And "If the password policies are set to be in conflict of each other you may not be able to change your password" any example for such a this situation?

Reply:
> And "If the password policies are set to be in conflict of each other you may not be able to change your password" any example for such a this situation?

Yes.
**Example scenario 1:**
- password policy regex enforcing no special characters
- user store manager patter enforcing to have at least some of special characters
Since no string satisfies both, validation would always fail.

**Example scenario 2:**
- Either of the regexes enforcing the password should be length `{5,7}`
- Min password length property is set to `8`
- Max password length property is set to `15`
Since no string can satisfy all the length conditions, validation would always fail.

Reply:
> I think u wanted to mean also the user store manager password pattern. Better to say as it.. "both password policies and user store manager password pattern ".

Fixed with: https://github.com/wso2/docs-apim/pull/1512/commits/2858d5787980c02936b3694c4abc44117db694ab